bookOfHb.md

```diff
@@ -0,0 +1,216 @@
+# The Book of Self-Repair (Heartbeat First Impressions)
+
+
+## As a developer, I want to successfully hack on Self-Repair
+
+### Setup (do once)
+
+1. Pots and Pans
+
+	```
+	# required: git, node
+	brew install node   # includes npm
+	brew install git    # git
+	
+	# optional
+	brew install hub    # github's drop in git client
+	brew install ghi    # github issues clien
+	```
+	
+
+1. Groceries
+
+	```
+	# usual clone and fork
+	
+	git clone mozilla/self-repair-server
+	cd self-repair-server
+	hub fork       
+	```
+
+1. Prep the veggies
+
+	```
+	# usual node package install
+	npm install   
+	# creates dependecies at `node_modules`
+	```
+
+1. Adjust the oven.
+
+	**Firefox Settings** - allow local debug in `about:config`
+	
+	- `browser.uitour.requireSecure;false`
+	- `browser.uitour.testingOrigins;http://localhost`
+
+
+### Modify 
+
+```
+ghi create -m "Some issue that need fixing"  
+git checkout -b 151-my-issue
+
+# edit files in `src/`
```

Comment:
that would maybe be helpful - probably doesn't need to be very in-depth.

On Thu, Sep 17, 2015 at 1:56 PM, Gregg Lind <notifications@github.com>
wrote:

> In bookOfHb.md
> <https://github.com/mozilla/self-repair-server/pull/151#discussion_r39797961>
> :
>
> > +
> > +1. Adjust the oven.
> > +
> > +	**Firefox Settings** - allow local debug in `about:config`
> > +	
> > +	- `browser.uitour.requireSecure;false`
> > +	- `browser.uitour.testingOrigins;http://localhost`
> > +
> > +
> > +### Modify
> > +
> > +```
> > +ghi create -m "Some issue that need fixing"
> > +git checkout -b 151-my-issue
> > +
> > +# edit files in `src/`
>
> Yes, those are utils copied from jetpack. I am trying to recall what was
> so aweosme to have from there :)
>
> Is 'explain the file tree' a section i need?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/self-repair-server/pull/151/files#r39797961>.
>